metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/ObjectsImporter.java

```diff
@@ -158,20 +238,38 @@ public Sample importSample(String accession) {
         return sampleAccessions;
     }
 
-    private Set<String> getReferenceSequenceAccessions(AnalysisType analysisType) {
-        Set<String> referenceSequenceAccessions = new HashSet<>();
-        ReferenceSequenceType referenceSequenceType = analysisType.getANALYSISTYPE().getREFERENCEALIGNMENT();
-        if (referenceSequenceType == null) {
-            return referenceSequenceAccessions;
-        }
-        ReferenceAssemblyType referenceAssemblyType = referenceSequenceType.getASSEMBLY();
-        if (referenceAssemblyType == null) {
-            return referenceSequenceAccessions;
+    private ReferenceSequence buildReferenceSequence(AnalysisType analysisType) {
+        ReferenceSequence referenceSequence = null;
+        AnalysisType.ANALYSISTYPE analysistype = analysisType.getANALYSISTYPE();
+        ReferenceSequenceType referenceSequenceType = null;
+
+        if (analysistype.isSetSEQUENCEVARIATION()) {
+            referenceSequenceType = analysistype.getSEQUENCEVARIATION();
+        } else if (analysistype.isSetREFERENCEALIGNMENT()) {
+            referenceSequenceType = analysistype.getREFERENCEALIGNMENT();
+        } else if (analysistype.isSetPROCESSEDREADS()) {
+            referenceSequenceType = analysistype.getPROCESSEDREADS();
         }
-        ReferenceAssemblyType.STANDARD standard = referenceAssemblyType.getSTANDARD();
-        if (standard != null) {
-            referenceSequenceAccessions.add(standard.getAccession());
+
+        if (referenceSequenceType != null) {
+            ReferenceAssemblyType referenceAssemblyType = referenceSequenceType.getASSEMBLY();
+            if (referenceAssemblyType != null) {
+                ReferenceAssemblyType.STANDARD standard = referenceAssemblyType.getSTANDARD();
```

Comment:
This all logic of manual creation of ReferenceSequnce should go to SraImporterDatabase

Reply:
This comment was probably made before @iyangar made his latest changes (I'm not sure). Anyway in the latest implementation `getAccessionFromReferenceSequenceType` method should remain in the `ObjectsImporter` class because it is used in both API and database cases. It is used to extract accession from ENA's `ReferenceSequenceType`.